File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
              List<Boolean> result = new ArrayList<>();
        int maxCandies = 0;

        for (int i : candies){
            maxCandies = Math.max(maxCandies,i);
        }
        for (int i : candies){
            if( i+extraCandies >= maxCandies){
                result.add(true);
            }else{
                result.add(false);
            }
        }


        return result;
    }
}

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
  maxCandies := 0
	result := make([]bool, len(candies))
	for _, i := range candies {
		maxCandies = max(maxCandies, i)
	}
	for i, v := range candies {
		if v+extraCandies >= maxCandies {
			result[i] = true
		} else {
			result[i] = false
		}
	}
	return result  
}